httpx_auth/_oauth2/common.py

```diff
@@ -86,3 +87,32 @@ def request_new_grant_with_post(
 class OAuth2:
     token_cache = TokenMemoryCache()
     display = DisplaySettings()
+
+
+class OAuth2BaseAuth(abc.ABC, httpx.Auth):
+    state: Optional[str] = None
+    early_expiry: float
+
+    refresh_token: Optional[Callable]
+
+    def auth_flow(
+        self, request: httpx.Request
+    ) -> Generator[httpx.Request, httpx.Response, None]:
+        token = OAuth2.token_cache.get_token(
+            self.state,
+            early_expiry=self.early_expiry,
+            on_missing_token=self.request_new_token,
+            on_expired_token=(
+                self.refresh_token if "refresh_token" in dir(self) else None
+            ),
```

Comment:
Remove this condition and rely on the fact that this is provided as parameter now